### ISSUE TITLE
 ammonite.integration.StandaloneTests.hello fails if predef.scala has output

### ISSUE THREAD

**Issue by ssanj** [lihaoyi/Ammonite#262]
When running the  ammonite.integration.StandaloneTests, I was getting an unusual error:

```
[info] 1/4     ammonite.integration.StandaloneTests.hello
[info]              utest.AssertionError: assert(evaled == "Hello World")
[info]              evaled: String = loading common dependencies...
[info]              loaded scalaz-core, scalaz-concurrent, joda-time
[info]              Hello World
[info] 2/4     ammonite.integration.StandaloneTests.complex
[info]              java.lang.RuntimeException: Nonzero exit code: 1
[info] 3/4     ammonite.integration.StandaloneTests.shell       Success
[info] 4/4     ammonite.integration.StandaloneTests     Success
[info] -----------------------------------Results-----------------------------------
[info] ammonite.integration.StandaloneTests     Success
[info]     hello        Failure('utest.AssertionError: assert(evaled == "Hello World")
[info]     evaled: String = loading common dependencies...
[info]     loaded scalaz-core, scalaz-concurrent, joda-time
[info]     Hello World')
[info]     complex      Failure('java.lang.RuntimeException: Nonzero exit code: 1')
[info]     shell        Success
[info] Failures:
[info] 1/4     ammonite.integration.StandaloneTests.hello
[info]              utest.AssertionError: assert(evaled == "Hello World")
[info]              evaled: String = loading common dependencies...
[info]              loaded scalaz-core, scalaz-concurrent, joda-time
[info]              Hello World
```

After a bit of digging around I figured out that the StandaloneTests where expecting:

```
Hello World
```

but received:

```
loading common dependencies...
loaded scalaz-core, scalaz-concurrent, joda-time
Hello World
```

Since the evaluation used == for the comparison it was failing:

```
'hello{
   val evaled = exec("Hello.scala")
   assert(evaled == "Hello World")
}
```

I had a look at the Hello.scala file:

```
println("Hello World")
```

and it did not have any dependencies. 

Where were these dependencies being loaded from?

It turned out that I had customised my ~/.ammonite/predef.scala file as:

```
println("loading common dependencies...")
load.ivy("org.scalaz" %% "scalaz-core" % "7.1.5")
load.ivy( "org.scalaz" %% "scalaz-concurrent" % "7.1.5")
load.ivy("org.scalacheck" %% "scalacheck" % "1.12.5")
load.ivy("joda-time" % "joda-time" % "2.9")
println("loaded scalaz-core, scalaz-concurrent, joda-time")
 
@
 
repl.prompt() = "amm> "
```

So basically if the ~/.Ammonite/predef.scala has been customised to output any text to the console the  ammonite.integration.StandaloneTests.hello will fail. A simple fix is to use:

```
    'hello{
      val evaled = exec("Hello.scala")
      assert(evaled.contains("Hello World"))
    }
```

I'm happy to submit a PR for this if you think it's an issue.

**Comment by ssanj** [comment]
Alternatively we need to prevent ~/.Ammonite/predef.scala being loaded when this test is run.

**Comment by lihaoyi** [comment]
yeah. In the readme I told people to start using `--predef-file ...` to set the predef but I wonder if there's an easier to way to hardcode a dev-default-predef and have it change for the published artifacts. Maybe checking if `ammonite.Constants.version.endsWith("-SNAPSHOT")`?
